core/peersholder/peersHolder.go

```diff
@@ -85,32 +87,23 @@ func (ph *peersHolder) Remove(peerID core.PeerID) {
 	ph.Lock()
 	defer ph.Unlock()
 
-	_, found := ph.peerIDs[peerID]
+	pidData, found := ph.peerIDs[peerID]
 	if !found {
 		return
 	}
 
 	shard, index, _ := ph.getShardAndIndexForPeer(peerID)
 	ph.removePeerFromMapAtIndex(shard, index)
 
-	delete(ph.peerIDs, peerID)
-
-	pubKeyForPeerID := ""
-	for pubKey, peerInfo := range ph.pubKeysToPeerIDs {
-		if peerInfo == nil {
-			continue
-		}
+	pubKey := pidData.pubKey
 
-		if peerInfo.pid == peerID {
-			pubKeyForPeerID = pubKey
-			break
-		}
-	}
+	delete(ph.peerIDs, peerID)
 
-	if len(pubKeyForPeerID) > 0 {
+	_, isPreferredPubKey := ph.pubKeysToPeerIDs[pubKey]
+	if isPreferredPubKey {
```

Comment:
kind of useless check since if we got to this point, it means that the provided pid was a preferred pid already.